docker/docker_image_src.go

```diff
@@ -305,53 +305,92 @@ func (s *dockerImageSource) GetBlobAt(ctx context.Context, info types.BlobInfo,
 		}
 		return nil, nil, err
 	}
-	if res.StatusCode != http.StatusPartialContent {
-		res.Body.Close()
-		return nil, nil, errors.Errorf("invalid status code returned when fetching blob %d (%s)", res.StatusCode, http.StatusText(res.StatusCode))
-	}
 
-	mediaType, params, err := mime.ParseMediaType(res.Header.Get("Content-Type"))
-	if err != nil {
-		return nil, nil, err
-	}
-
-	streams := make(chan io.ReadCloser)
-	errs := make(chan error)
+	switch res.StatusCode {
+	case http.StatusOK:
+		// if the server replied with a 200 status code, convert the full body response to a series of
+		// streams as it would have been done with 206.
+		streams := make(chan io.ReadCloser)
+		errs := make(chan error)
+		go func() {
```

Comment:
This is becoming rather large (e.g. how far down the “unknown `res.StatusCode` case is); please split the two `streams`-producing goroutines into two separate functions.

Unit tests also look worthwhile, at least for those `streams` producers; without testing actual pulls of this format with high frequency (the way Skopeo and Podman tests trigger at least the default pull paths), we are pretty unlikely to notice breakage.